nutanix/resource_nutanix_virtual_machine.go

```diff
@@ -1203,6 +1203,12 @@ func resourceNutanixVirtualMachineUpdate(ctx context.Context, d *schema.Resource
 		metadata.SpecVersion = specVersion
 	}
 
+	// Check if we are dealing with SECURE_BOOT machine, MachineType is needed then set to Q35
```

Comment:
we don't need this check at two places. we can always preFill the machineType using the response.
here: https://github.com/nutanix/terraform-provider-nutanix/blob/033e0f76532dc2c8df09befa355e35798b4db6d4/nutanix/resource_nutanix_virtual_machine.go#L928